mmf_product.cpp

```diff
@@ -0,0 +1,192 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <fstream>
+#include <sstream>
+
+std::vector<double> readMMformat_vector(const std::string & filename){
```

Comment:
Screens are wide these days, I would suggest making your name more explanatory. Without context the user would have no idea that `MM` means matrix_market.